index.ts

```diff
@@ -1,27 +1,17 @@
+type BaseType = 'string' | 'number' | 'bigint' | 'boolean' | 'symbol' | 'undefined' | 'object' | 'function';
+type RealType = string;
+type CountRealTypes = {
+    [key: RealType]: number;
+};
+
 // Test utils
 
-const testBlock = (name) => {
+const testBlock = (name: string): void => {
     console.groupEnd();
     console.group(`# ${name}\n`);
 };
 
-/**
- * Функция сравнивает два массива с заданной глубиной вложенности.
- * @param {Array} a Массив.
- * @param {Array} b Массив.
- * @param {number} [maxDepth=5] Максимальная допустимая глубина вложенности сравниваемых массивов.
- * @returns {boolean} Возвращает true если массивы равны и false если нет.
- * @throw Выбрасывает ошибку если аргумент a или b не являются массивами.
- * @throw Выбрасывает ошибку если maxDepth не число.
- * @throw Выбрасывает ошибку если превышена максимальная глубина вложенности в переданных массивах.
- */
-const compareTwoArrays = (a, b, maxDepth = 5) => {
-    if (!Array.isArray(a) || !Array.isArray(b)) {
-        throw new Error('One of the arguments is not an array!');
-    }
-    if (typeof maxDepth !== 'number' || Number.isNaN(maxDepth)) {
-        throw new Error('Maximum depth is not a number!');
-    }
+const compareTwoArrays = (a: any[], b: any[], maxDepth = 5): boolean => {
```

Comment:
@Beken-S Почему для аргументов выбрал тип `any`?

Reply:
Функция должна сравнивать любые массивы, поэтому подумал, что это допустимо в данном случае.

Reply:
```suggestion
const compareTwoArrays = (a: unknown[], b: unknown[], maxDepth = 5): boolean => {
    if (maxDepth < 0) {
        throw new Error('Maximum depth exceeded!');
    }
    if (a === b) {
        return true;
    }
    if (a.length !== b.length) {
        return false;
    }

    for (let i = 0; i < a.length; i++) {
        const itemA: unknown = a[i];
        const itemB: unknown = b[i];
        if (Array.isArray(itemA) && Array.isArray(itemB)) {
            if (!compareTwoArrays(itemA, itemB, maxDepth - 1)) {
                return false;
            }
        } else if (itemA !== itemB) {
            return false;
        }
    }
    return true;
};
```
Возможно так будет лучше?